galsim/phase_psf.py

```diff
@@ -0,0 +1,1442 @@
+
+        @returns  stepK.
+        """
+        # Generically, Galsim propagates stepK() for convolutions using
+        #   stepk = sum(s**-2 for s in stepks)**(-0.5)
+        # We're not actually doing convolution between screens here, though.  In fact, the right
+        # relation for Kolmogorov screens uses exponents -5./3 and -3./5:
+        #   stepk = sum(s**(-5./3) for s in stepks)**(-3./5)
+        # Since most of the layers in a PhaseScreenList are likely to be (nearly) Kolmogorov
+        # screens, we'll use that relation.
+        return sum(layer.stepK(**kwargs)**(-5./3) for layer in self)**(-3./5)
+
+
+class PhaseScreenPSF(GSObject):
+    """A PSF surface brightness profile constructed by integrating over time the instantaneous PSF
+    derived from a set of phase screens and an aperture.
+
+    There are at least three ways to construct a PhaseScreenPSF given a PhaseScreenList.  The
+    following two statements are equivalent:
+        >>> psf = screen_list.makePSF(...)
+        >>> psf = PhaseScreenPSF(screen_list, ...)
+
+    The third option is to use screen_list.makePSF() to obtain multiple PSFs at different field
+    angles from the same PhaseScreenList over the same simulated time interval.  Depending on the
+    details of PhaseScreenList and other arguments, this may be significantly faster than manually
+    looping over makePSF().
+        >>> psfs = screen_list.makePSF(..., theta_x=[...], theta_y=[...])
+
+    Note that constructing a PhaseScreenPSF advances each PhaseScreen in `screen_list`, so PSFs
+    constructed consecutively with the same arguments will generally be different.  Use
+    PhaseScreenList.reset() to reset the time to t=0.
+
+        >>> screen_list.reset()
+        >>> psf1 = screen_list.makePSF(...)
+        >>> psf2 = screen_list.makePSF(...)
+        >>> assert psf1 != psf2
+        >>> screen_list.reset()
+        >>> psf3 = screen_list.makePSF(...)
+        >>> assert psf1 == psf3
+
+    Computing a PSF from a phase screen also requires an Aperture be specified.  This can be done
+    either directly via the `aper` keyword, or by setting a number of keywords that will be passed
+    to the `Aperture.fromPhaseScreenList` constructor.  The `aper` keyword always takes precedence.
+
+    @param screen_list       PhaseScreenList object from which to create PSF.
+    @param lam               Wavelength in nanometers at which to compute PSF.
+    @param exptime           Time in seconds overwhich to accumulate evolving instantaneous PSF.
+                             [Default: 0.0]
+    @param flux              Flux of output PSF [Default: 1.0]
+    @param theta_x           x-component of field angle at which to evaluate phase screens and
+                             resulting PSF.  [Default: 0.0*galsim.arcmin]
+    @param theta_y           y-component of field angle at which to evaluate phase screens and
+                             resulting PSF.  [Default: 0.0*galsim.arcmin]
```

Comment:
Some more `Default` -> `default` here.  Also, for parallelism, wouldn't it make sense to allow `theta` here as you did for `makePSF` above?